data/refactored_example_realization_config.json

```diff
@@ -0,0 +1,151 @@
+{
+    "global": {
+        "tshirt": {
+            "maxsmc": 0.439,
+            "wltsmc": 0.066,
+            "satdk": 0.00000338,
+            "satpsi": 0.355,
+            "slope": 1.0,
+            "scaled_distribution_fn_shape_parameter": 4.05,
+            "multiplier": 0.0,
+            "alpha_fc": 0.33,
+            "Klf": 0.01,
+            "Kn": 0.03,
+            "nash_n": 2,
+            "Cgw": 0.01,
+            "expon": 6.0,
+            "max_groundwater_storage_meters": 1.0,
+            "nash_storage": [
+                0.0,
+                0.0
+            ],
+            "soil_storage_percentage": 0.667,
+            "groundwater_storage_percentage": 0.5,
+            "timestep": 3600,
+            "giuh": {
+                "giuh_path": "./test/data/giuh/GIUH.json",
+                "crosswalk_path": "./data/crosswalk.json"
+            }
+        },
+        "forcing": {
+            "file_pattern": ".*{{ID}}.*.csv", 
+            "path": "./data/forcing/",
+            "start_time": "2015-12-01 00:00:00", 
+            "end_time": "2015-12-30 23:00:00"
+        }
+    },
+    "catchments": {
+        "cat-87": {
+            "tshirt": {
+                "maxsmc": 0.439,
+                "wltsmc": 0.066,
+                "satdk": 0.00000338,
+                "satpsi": 0.355,
+                "slope": 1.0,
+                "scaled_distribution_fn_shape_parameter": 4.05,
+                "multiplier": 0.0,
+                "alpha_fc": 0.33,
+                "Klf": 0.01,
+                "Kn": 0.03,
+                "nash_n": 2,
+                "Cgw": 0.01,
+                "expon": 6.0,
+                "max_groundwater_storage_meters": 1.0,
```

Comment:
I would agree that representing data objects in json in manner that is closer to the c++ object layout would be beneficial. In fact it maybe worth the time to create formal interface for json serialization and de-serialization that should implemented by all data objects and then have the parsing code just call de-serialize on the input stream after the type identifier is parsed.

Reply:
Not sure why crappy noah-mp parameter names are making it into our framework.  Not good.  Why?   They are nondescript and vague without context or units.  Suggest those go away and are replaced with meaning fulnames.  Happy to suggest some if you ask.

Reply:
The parameters that are in the config are just the ones in the functions themselves. The idea is that the parser reads in the JSON tree, then the formulations pick their values from the tree after everything has been created. So, a tshirt formulation is needed, it's created, then the parameters that it will feed to the actual model are grabbed from the tree. Currently, tshirt requires a parameter named `expon`, so it will reach into the tree and try to get a value for `expon`. Now, if that parameter in code changed to `expon_i_dunno_im_not_the_one_with_a_masters_or_doctorate`, that should also be reflected within the config as well. It's the config's job to bear data and it's the formulation's job to retrieve what it needs from it.

I think it's great news that those parameter names are crappy; it means that there seems to be a few fairly obvious places where the code can be improved. That probably means that `tshirt_params`, `tshirt_realization`, `tshirt_state`, etc, can all see some improvements.

Reply:
> Not sure why crappy noah-mp parameter names are making it into our framework. Not good. Why? They are nondescript and vague without context or units. Suggest those go away and are replaced with meaning fulnames. Happy to suggest some if you ask.

I would suggest starting here and elaborating on the names in the documentation that those components were originally derived from.  We should be able to look at the params/state structs and get a good sense of how to include this feedback into this original document.

https://github.com/NOAA-OWP/ngen/blob/master/doc/T-shirt_model_description.pdf